interpreter/llvm-project/clang/lib/Basic/SourceManager.cpp

```diff
@@ -2095,10 +2095,7 @@ bool SourceManager::isBeforeInTranslationUnit(SourceLocation LHS,
       return LIsScratch;
     return LOffs.second < ROffs.second;
   }
-  //AXEL: Work around diags from include chains not rooted in main file.
-  //AXEL: llvm_unreachable("Unsortable locations found");
-  assert(0 && "Unsortable locations found");
-  return LOffs.first < ROffs.first;
+  llvm_unreachable("Unsortable locations found");
```

Comment:
Thiw would corrupt the stack if it reaches here. That's the problem can be triggered by generated code.

Reply:
Yes, I see... So you prefer to keep this patch for now?

Reply:
I am not sure. I believe that the `initializeVirtualFile` is not sufficient in the cases where we generate code (via the ast synthesizers) where we pass invalid source locations. However, it might be worth to try. Let's ask cmssw.